opencensus/metrics/export/gauge.py

```diff
@@ -0,0 +1,222 @@
+    :class:`LongGauge` for a given set of label values.
+    """
+
+    def __init__(self):
+        self.value = 0
+        self._value_lock = threading.Lock()
+
+    def __repr__(self):
+        return ("{}({})"
+                .format(
+                    type(self).__name__,
+                    self.value
+                ))
+
+    def add(self, val):
+        """Add `val` to the current value.
+
+        :type val: int
+        :param val: Value to add.
+        """
+        if not isinstance(val, six.integer_types):
+            raise ValueError("GaugePointLong only supports integer types")
+        with self._value_lock:
+            self.value += val
+
+    def set(self, val):
+        """Set the current value to `val`.
+
+        :type val: int
+        :param val: Value to set.
+        """
+        if not isinstance(val, six.integer_types):
+            raise ValueError("GaugePointLong only supports integer types")
+        with self._value_lock:
+            self.value = val
+
+
+class GaugePointDouble(object):
+    """An instantaneous measurement from a DoubleGauge.
+
+    A GaugePointDouble represents the most recent measurement from a
+    :class:`DoubleGauge` for a given set of label values.
+    """
+
+    def __init__(self):
+        self.value = 0.0
+        self._value_lock = threading.Lock()
+
+    def __repr__(self):
+        return ("{}({})"
+                .format(
+                    type(self).__name__,
+                    self.value
+                ))
+
+    def add(self, val):
+        """Add `val` to the current value.
+
+        :type val: float
+        :param val: Value to add.
+        """
+        with self._value_lock:
+            self.value += val
+
+    def set(self, val):
+        """Set the current value to `val`.
+
+        :type val: float
+        :param val: Value to set.
+        """
+        with self._value_lock:
+            self.value = float(val)
+
+
+class Gauge(object):
+    """A set of instantaneous measurements of the same type.
+
+    End users should use :class:`LongGauge` or :class:`DoubleGauge` instead of
+    using this class directly.
+
+    The constructor arguments are used to create a
+    :class:`opencensus.metrics.export.metric_descriptor.MetricDescriptor` for
+    converted metrics. See that class for details.
+    """
+
+    def __init__(self, name, description, unit, label_keys):
+        self._len_label_keys = len(label_keys)
+        self.descriptor = metric_descriptor.MetricDescriptor(
+            name, description, unit, self.descriptor_type, label_keys)
+        self.points = OrderedDict()
+        self._points_lock = threading.Lock()
+
+    def __repr__(self):
+        return ('{}(descriptor.name="{}", points={})'
+                .format(
+                    type(self).__name__,
+                    self.descriptor.name,
+                    self.points
+                ))
+
+    def get_time_series(self, label_values):
```

Comment:
I think is ```getDefaultTimeSeries ``` missing. https://github.com/census-instrumentation/opencensus-java/blob/master/api/src/main/java/io/opencensus/metrics/DoubleGauge.java#L104

Reply:
That's true, I didn't understand the rationale for the default time series in the java client. The goal is that there's a single default point (read: time series) for each gauge that isn't associated with any label values?

Reply:
Added in a2e56cf, let me know what you think.